### PR TITLE
Added Chapter 10

### PR DESCRIPTION
**Pull request by MarkWignall**

### CODE REVIEW

File: chapter10/mdw/Ex-02.pl [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use feature "say";
+use Scalar::Util qw(looks_like_number);
+
+my $log; ##Add = 1 to turn off logging
```
Comment (codavex) [inline chapter10/mdw/Ex-02.pl:8]: So you're logging if $log isn't true?

Comment (MarkWignall) [inline chapter10/mdw/Ex-02.pl:8]: Yeah, not the most intuitive naming of variables.

File: chapter10/mdw/Ex-01.pl [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	$attempts += 1;
+	if ( $guess == $number ){
+		if ( $attempts == 1 ){
+			say "You won with your first guess!";
+		} else {
+			say "You Win. It took " . $attempts . " guesses";
+		}
+		last;
+	}
+	if ( $guess > $number ){
```
Comment (codavex) [inline chapter10/mdw/Ex-01.pl:32]: you could elsif these together for slight efficiency increase

File: chapter10/mdw/Ex-01.pl [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+use Scalar::Util qw(looks_like_number);
+
+my $attempts = 0;
+my $number = int(1 + rand 100);
+
+say "Welcome, try to guess my number";
+
+while (<>){
+	my $guess = $_;
+	if ( $guess =~ /exit|quit/){
```
Comment (codavex) [inline chapter10/mdw/Ex-01.pl:15]: empty line should exit the loop too

File: chapter10/mdw/Ex-01.pl [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+		say "You gave up after " . $attempts . " tries";
+		last;
+	}
+	if ( !looks_like_number( $guess ) ){
+		say "Thats not a number. You lose.";
```
Comment (codavex) [inline chapter10/mdw/Ex-01.pl:20]: Harsh.

But fair.

File: chapter10/mdw/Ex-01.pl [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+		last;
+	}
+	$attempts += 1;
```
Comment (codavex) [inline chapter10/mdw/Ex-01.pl:23]: Feels wrong to be incrementing in the middle of the loop

**Pull request merged**